tests/unittests/test_timeout.py

```diff
@@ -0,0 +1,747 @@
+        Context.config = {'request_timeout': 0.0}
+        Context.asana.client.stories = mock.MagicMock()
+        Context.asana.client.stories.get_stories_for_task.side_effect = valid_data
+
+        stories_for_tasks = stories.get_stories_for_tasks('test', ['test'])
+
+        # verify if the timeout was passed as expected from the config file
+        Context.asana.client.stories.get_stories_for_task.assert_called_with(opt_fields=['test'], task_gid='test', timeout=300.0)
+
+    def test_timeout_string_0_value_in_config__get_stories_for_tasks(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        # initialize config file
+        Context.config = {'request_timeout': '0.0'}
+        Context.asana.client.stories = mock.MagicMock()
+        Context.asana.client.stories.get_stories_for_task.side_effect = valid_data
+
+        stories_for_tasks = stories.get_stories_for_tasks('test', ['test'])
+
+        # verify if the timeout was passed as expected from the config file
+        Context.asana.client.stories.get_stories_for_task.assert_called_with(opt_fields=['test'], task_gid='test', timeout=300.0)
+
+@mock.patch("tap_asana.asana.Asana.refresh_access_token")
+@mock.patch("time.sleep")
+class TestTimeoutErrorTeams(unittest.TestCase):
+
+    def test_timeout_error__find_team_by_organization(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        Context.asana.client.teams = mock.MagicMock()
+        Context.asana.client.teams.find_by_organization.side_effect = raise_Timeout_error
+
+        try:
+            team_by_organization = teams.find_team_by_organization('test', ['test'])
+        except requests.Timeout:
+            pass
+
+        self.assertEqual(Context.asana.client.teams.find_by_organization.call_count, 5)
+
+    def test_no_timeout_error__find_team_by_organization(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        Context.asana.client.teams = mock.MagicMock()
+        Context.asana.client.teams.find_by_organization.side_effect = valid_data
+
+        team_by_organization = teams.find_team_by_organization('test', ['test'])
+
+        self.assertEqual(team_by_organization, list(valid_data()))
+
+    def test_timeout_error__get_users_for_teams(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        Context.asana.client.teams = mock.MagicMock()
+        Context.asana.client.teams.users.side_effect = raise_Timeout_error
+
+        try:
+            users_for_team = teams.get_users_for_teams('test')
+        except requests.Timeout:
```

Comment:
Please explain the reason for the exception here. Mention it is intentional.

Reply:
Added comment.